org.scala-ide.sdt.core/src/org/scalaide/core/completion/ScalaCompletions.scala

```diff
@@ -25,11 +25,21 @@ import org.scalaide.util.internal.Commons
 class ScalaCompletions extends HasLogger {
   import org.eclipse.jface.text.IRegion
 
+  @deprecated("Use getCompletions, which correctly handles units with a different presentation compiler", since = "4.1.0")
+  def findCompletions(region: IRegion, position: Int, icu: InteractiveCompilationUnit): List[CompletionProposal] =
+    icu.scalaProject.presentationCompiler { compiler =>
+      findCompletion(region, position, icu, compiler)
+    }.getOrElse(Nil)
+
+
   /** Returns the list of possible completions, at the given position in the compilation unit,
    *  with the given region as completion prefix.
+   *
+   *  @note This supports compilation units that have a different presentation compiler than the default
+   *        per-project compiler
    */
-  def findCompletions(region: IRegion, position: Int, icu: InteractiveCompilationUnit): List[CompletionProposal] =
-    icu.scalaProject.presentationCompiler { compiler =>
+  def getCompletions(region: IRegion, position: Int, icu: InteractiveCompilationUnit): List[CompletionProposal] =
```

Comment:
Because of binary incompatibilities. The new method in `InteractiveCompilationUnit` is not going to be available on subclasses that compiled against the 4.0 version. So we can't add a call to the new method on an old code path, or clients will see `NoSuchMethodErrors`.